qadence/backends/pyqtorch/backend.py

```diff
@@ -83,42 +89,18 @@ def run(
         unpyqify_state: bool = True,
     ) -> Tensor:
         n_qubits = circuit.abstract.n_qubits
-
-        if state is not None:
-            if pyqify_state:
-                if (state.ndim != 2) or (state.size(1) != 2**n_qubits):
-                    raise ValueError(
-                        "The initial state must be composed of tensors of size "
-                        f"(batch_size, 2**n_qubits). Found: {state.size() = }."
-                    )
-
-                # PyQ expects a column vector for the initial state
-                # where each element is of dim=2.
-                state = state.T.reshape([2] * n_qubits + [state.size(0)])
-            else:
-                if prod(state.size()[:-1]) != 2**n_qubits:
-                    raise ValueError(
-                        "A pyqified initial state must be composed of tensors of size "
-                        f"(2, 2, ..., batch_size). Found: {state.size() = }."
-                    )
+        if state is None:
+            # If no state is passed, we infer the batch_size through the length
+            # of the individual parameter value tensors.
+            state = circuit.native.init_state(batch_size=infer_batchsize(param_values))
         else:
-            # infer batch_size without state
-            if len(param_values) == 0:
-                batch_size = 1
-            else:
-                batch_size = max([len(tensor) for tensor in param_values.values()])
-            state = circuit.native.init_state(batch_size=batch_size)
+            # pyqtorch expects input shape [2] * n_qubits + [batch_size]
+            state = (
+                pyqify(state, n_qubits) if pyqify_state else validate_and_convert(state, n_qubits)
```

Comment:
What does the `validate_and_convert` method do exactly?

Reply:
i renamed it to "validate_state" and it only verifies that a given input is either of qadence shape (batch_size, 2**n_qubits) or pyqtorch shape ([2] * n_qubits + [batch_size]). there is a docstring